core/new-gui/src/app/workspace/component/input-autocomplete/input-autocomplete.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit } from "@angular/core";
+import { FieldType } from "@ngx-formly/core";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { UserFileService } from "src/app/dashboard/service/user-file/user-file.service";
+import { FormControl } from "@angular/forms";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-input-autocomplete-template",
+  templateUrl: "./input-autocomplete.component.html",
+  styleUrls: ["input-autocomplete.component.scss"],
+})
+
+/* *
+ * The FieldType<any> is a workaround for the issue of not assignable FormControl.
+ * details https://github.com/ngx-formly/ngx-formly/issues/2842#issuecomment-1066116865
+ * need to upgrade formly to v6 to properly fix this issue.
+ */
+export class InputAutoCompleteComponent extends FieldType<any> implements OnInit {
+  inputValue: string = "";
+  title?: string;
+  // the autocomplete selection list
+  public selections: string[] = [];
+  fullList: string[] = [];
```

Comment:
the `fullList` seems to be only a local variable in that function. don't declare here. Also please change it to a better name: `fullList` is not quite meaningful.

Reply:
solved